File: MVC_O_DiceGame/DiceGame_ruby/controller/Player.rb
module Controller
  class Player

    def initialize a_view
      @view = a_view;
    end

    def play a_game
      a_game.add_subscriber self 
      @view.display_instructions

      until @view.wants_to_quit? do
        @view.display_instructions
        @view.display_result a_game.play
      end
    end

    def rolled_dice face_value
      @view.display_dice_value face_value
    end

  end
end

File: MVC_O_DiceGame/MVC_O_DiceGame_Swift_3.0.1/MVC_O_DiceGame_Swift_3.0.1/controller/Player.swift
//
//  Player.swift
//  MVC_O_DiceGame_Swift_3.0.1
//
//  Created by Sunny Buttar on 2016-11-13.
//  Copyright © 2016 Sarpreet Singh. All rights reserved.
//

import Foundation

public class Player : IDiceRolledObserver {
    
    private var m_view : IView;
    
    init(a_view : IView) {
        m_view = a_view;
    }
    
    func playGame(a_game : DiceGame) {
        
        a_game.addSubscriber(a_sub: self);
        m_view.displayWelcomeMessage();
        m_view.displayInstructions();
        
        while m_view.wantsToPlay() {
            m_view.displayResult(a_win: a_game.play());
            
            m_view.displayInstructions();
        }
    }
    
    func diceRolled(a_faceValue: Int) {
        m_view.displayDiceValue(a_value: a_faceValue);
    }
}
